graylog2-server/src/main/java/org/graylog2/security/realm/LdapUserAuthenticator.java

```diff
@@ -159,12 +141,83 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken authtok
         return null;
     }
 
+    protected LdapNetworkConnection openLdapConnection(LdapSettings ldapSettings) throws LdapException {
+        final LdapConnectionConfig config = new LdapConnectionConfig();
+        config.setLdapHost(ldapSettings.getUri().getHost());
+        config.setLdapPort(ldapSettings.getUri().getPort());
+        config.setUseSsl(ldapSettings.getUri().getScheme().startsWith("ldaps"));
+        config.setUseTls(ldapSettings.isUseStartTls());
+        if (ldapSettings.isTrustAllCertificates()) {
+            config.setTrustManagers(new TrustAllX509TrustManager());
+        }
+        config.setName(ldapSettings.getSystemUserName());
+        config.setCredentials(ldapSettings.getSystemPassword());
+        return ldapConnector.connect(config);
+    }
+
+    protected LdapEntry searchLdapUser(LdapNetworkConnection connection, String principal, LdapSettings ldapSettings) throws LdapException, CursorException {
+        return ldapConnector.search(connection,
+                ldapSettings.getSearchBase(),
+                ldapSettings.getSearchPattern(),
+                ldapSettings.getDisplayNameAttribute(),
+                principal,
+                ldapSettings.isActiveDirectory(),
+                ldapSettings.getGroupSearchBase(),
+                ldapSettings.getGroupIdAttribute(),
+                ldapSettings.getGroupSearchPattern());
+    }
+
     public boolean isEnabled() {
         final LdapSettings ldapSettings = ldapSettingsService.load();
         return ldapSettings != null && ldapSettings.isEnabled();
     }
 
     @Nullable
+    public User syncLdapUser(String principal) {
+
+        final LdapSettings ldapSettings = ldapSettingsService.load();
+        if (ldapSettings == null || !ldapSettings.isEnabled()) {
```

Comment:
This could be simplified to:
```
if (!isEnabled()) {
    // ...
}
```

Reply:
Agreed, I tried to follow the same style as in doGetAuthenticationInfo - would you be ok if I changed it in both places?

Reply:
@gaspardpetit Yes, that would be great! 👍